### PR TITLE
add wrapper for VoteInOrganizationScheme

### PR DESCRIPTION
**Pull request by dkent600**
Further to #24

### CODE REVIEW

File: lib/contracts/voteInOrganizationScheme.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+"use strict";
+const dopts = require("default-options");
```
Comment (tibetsprague) [inline lib/contracts/voteInOrganizationScheme.js]: by the way, why not start using typescript for new wrappers?

File: lib/custom_typings/system.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+declare module 'system' {
```
Comment (tibetsprague) [inline lib/custom_typings/system.d.ts]: why do we need this?

File: test/helpers.js [hunk 3]
```diff
@@ -1,16 +1,5 @@
...
-    helpers for tests
-*/
 import { Utils } from "../lib/utils.js";
 import { assert } from "chai";
-
-beforeEach(async () => {
-  global.web3 = Utils.getWeb3();
-  global.assert = assert;
-  global.accounts = [];
-  await etherForEveryone();
```
Comment (tibetsprague) [inline test/helpers.js]: we dont need this anymore?

File: test/schemeregistrar.js [hunk 4]
```diff
@@ -69,6 +69,7 @@ describe("SchemeRegistrar", () => {
     const dao = await forgeDao();
 
     const schemeRegistrar = await dao.getScheme("SchemeRegistrar");
+    // schemeRegistrar can't remove a scheme wit greater permissions that it has
```
Comment (tibetsprague) [inline test/schemeregistrar.js]: wit = with

File: tsconfig.json [hunk 5]
```diff
@@ -7,7 +7,8 @@
     "sourceRoot": "lib",
     "sourceMap": true,
     "baseUrl": "lib",
-    "allowJs": true
+    "allowJs": true,
+    // "checkJs": true
```
Comment (tibetsprague) [inline tsconfig.json]: lets make a decision on this

File: lib/contracts/voteInOrganizationScheme.js (hunk 1 above)
Comment (dkent600) [reply lib/contracts/voteInOrganizationScheme.js]: I created an issue  just for that.  I can't be sure there won't be issues with tools and packaging, so want to be able to concentrate on that.

File: lib/custom_typings/system.d.ts (hunk 2 above)
Comment (dkent600) [reply lib/custom_typings/system.d.ts]: This addresses some typings issues that will come up when we do convert to typescript.

File: test/helpers.js (hunk 3 above)
Comment (dkent600) [reply test/helpers.js]: It existed in two places, was duplicated.

File: tsconfig.json (hunk 5 above)
Comment (dkent600) [reply tsconfig.json]: Can't leave it in right now.  There still occurs one class of errors that I couldn't figure out how to resolve and didn't want to spend more time on.

File: test/schemeregistrar.js (hunk 4 above)
Comment (dkent600) [reply test/schemeregistrar.js]: Fixed

File: test/voteInOrganizationScheme.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+  return { proposalId: result.proposalId, votingMachine: votingMachine.address };
+};
+
+describe("VoteInOrganizationScheme", () => {
+  let dao;
+  let voteInOrganizationScheme;
+  beforeEach(async () => {
+
+    dao = await DAO.new({
```
Comment (orenyodfat) [inline test/voteInOrganizationScheme.js]: This code(DAO.new) is duplicate at vestingScheme (and probably on other places) .
Should it be moved to a common place (helpers.js ).

File: test/voteInOrganizationScheme.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    /**
+         * cast a vote using voteInOrganizationScheme's voting machine.
+         * assert that the proposal is executed.
+         */
+    const tx = await votingMachine.vote(result.proposalId, 1, { from: web3.eth.accounts[0] });
+    // TODO: Update these to use ProposalExecuted
+    // const eventAvatar = Utils.getValueFromLogs(tx, "_avatar", "LogExecuteProposal", 1);
+    // assert.equal(eventAvatar, dao.avatar.address);
+    const eventProposal = Utils.getValueFromLogs(tx, "_proposalId", "LogExecuteProposal", 1);
+    assert.equal(eventProposal, result.proposalId);
```
Comment (orenyodfat) [inline test/voteInOrganizationScheme.js]: Is that assert the voting in the originalIntVote ?

File: test/voteInOrganizationScheme.js (hunk 6 above)
Comment (dkent600) [reply test/voteInOrganizationScheme.js]: There is potential in that, but it is not the subject of this PR.

File: test/voteInOrganizationScheme.js [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+    assert.isOk(votingMachine);
+
+    /**
+         * cast a vote using voteInOrganizationScheme's voting machine.
+         * assert that the proposal is executed.
+         */
+    const tx = await votingMachine.vote(result.proposalId, 1, { from: web3.eth.accounts[0] });
+    // TODO: Update these to use ProposalExecuted
+    // const eventAvatar = Utils.getValueFromLogs(tx, "_avatar", "LogExecuteProposal", 1);
```
Comment (orenyodfat) [inline test/voteInOrganizationScheme.js]: Please remove commented code

Comment (dkent600) [reply test/voteInOrganizationScheme.js]: Done

File: test/voteInOrganizationScheme.js (hunk 7 above)
Comment (dkent600) [reply test/voteInOrganizationScheme.js]: No, it is the current dao.

Comment (orenyodfat) [reply test/voteInOrganizationScheme.js]: So what does it test ?

Comment (dkent600) [reply test/voteInOrganizationScheme.js]: I'm going to try filtering VoteProposal events for the expected event.

Comment (orenyodfat) [reply test/voteInOrganizationScheme.js]: Great. It will give the test a meaning :)

Comment (dkent600) [reply test/voteInOrganizationScheme.js]: @orenyodfat 

Thank you for spotting that I hadn't run a test on this.

I am finding that the voter on the original dao's proposal is accounts[0], the current user.

Wouldn't it make sense for that vote to have been done on behalf of the current DAO's avatar?

File: test/voteInOrganizationScheme.js (hunk 6 above)
Comment (dkent600) [reply test/voteInOrganizationScheme.js]: I am having too many issues with PRs that balloon when I have other PRs waiting behind that have to be merged, as well as PRs taking a long time to get approved.  Refactoring test code can easily and be done better in isolation.  This PR is about adding a wrapper.  How about we do a separate one on cleaning up test code (an ongoing process anyways...)?

File: test/voteInOrganizationScheme.js (hunk 7 above)
Comment (orenyodfat) [reply test/voteInOrganizationScheme.js]: why not, a user can be a member in multiple orgs.

Comment (orenyodfat) [reply test/voteInOrganizationScheme.js]: this tests suppose to auto run by travis.  are they?.

File: tsconfig.json (hunk 5 above)
Comment (dkent600) [reply tsconfig.json]: Regarding the remaining errors: I have confirmed they will go away when we move to TS.

**Pull request merged**